File: app/src/androidTest/java/com/github/romainlogean/sdp_bootcamp/MainActivityTest.java
package com.github.romainlogean.sdp_bootcamp;

import static androidx.test.espresso.Espresso.onView;
import static androidx.test.espresso.action.ViewActions.clearText;
import static androidx.test.espresso.action.ViewActions.click;
import static androidx.test.espresso.action.ViewActions.closeSoftKeyboard;
import static androidx.test.espresso.action.ViewActions.typeText;
import static androidx.test.espresso.intent.Intents.intended;
import static androidx.test.espresso.intent.matcher.IntentMatchers.hasComponent;
import static androidx.test.espresso.intent.matcher.IntentMatchers.hasExtra;

import static org.hamcrest.core.AllOf.allOf;

import androidx.test.espresso.intent.Intents;
import androidx.test.espresso.matcher.ViewMatchers;
import androidx.test.ext.junit.rules.ActivityScenarioRule;
import androidx.test.ext.junit.runners.AndroidJUnit4;

import org.junit.Rule;
import org.junit.Test;
import org.junit.runner.RunWith;

@RunWith(AndroidJUnit4.class)
public class MainActivityTest {
    @Rule
    public ActivityScenarioRule<MainActivity> testRule = new ActivityScenarioRule<>(MainActivity.class);

    @Test
    public void ChangeActivityWithExtraTest(){

        Intents.init();

        onView(ViewMatchers.withId(R.id.mainName))
                .perform(clearText())
                .perform(typeText("Romain"))
                .perform(closeSoftKeyboard());

        onView(ViewMatchers.withId(R.id.mainGoButton)).perform(click());

        intended(allOf(hasExtra("NAME","Romain"),hasComponent(GreetingActivity.class.getName())));

        Intents.release();
    }
}

File: BootcampKotlin/app/src/androidTest/java/com/github/ybecker/bootcampKotlin/MainActivityTest.kt
package com.github.ybecker.bootcampKotlin

import androidx.test.espresso.action.ViewActions
import androidx.test.espresso.intent.Intents
import androidx.test.espresso.matcher.ViewMatchers
import androidx.test.ext.junit.rules.ActivityScenarioRule
import androidx.test.ext.junit.runners.AndroidJUnit4

import org.junit.Rule
import org.junit.Test
import org.junit.runner.RunWith

import androidx.test.espresso.Espresso.onView
import androidx.test.espresso.intent.Intents.intended
import androidx.test.espresso.intent.matcher.IntentMatchers.*
import org.hamcrest.Matchers.allOf

@RunWith(AndroidJUnit4::class)
class MainActivityTest {
    @get:Rule
    val testRule = ActivityScenarioRule(MainActivity::class.java)

    @Test
    fun checkTransitionToGreetActivity() {
        Intents.init()

        val textInteraction = onView(ViewMatchers.withId(R.id.mainName))
        textInteraction.perform(
            ViewActions.clearText(),
            ViewActions.typeText("Yann"),
            ViewActions.closeSoftKeyboard()
        )

        val buttonInteraction = onView(ViewMatchers.withId(R.id.mainGoButton))
        buttonInteraction.perform(ViewActions.click())

        intended(allOf(
            hasExtra("name", "Yann"),
            hasComponent(GreetingActivity::class.java.name))
        )

        Intents.release()
    }
}
